src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,185 @@
+ * Random Testing" by Ma et. al (appears in ASE 2015):
+ * https://people.kth.se/~artho/papers/lei-ase2015.pdf .
+ *
+ * <p>Biases input selection towards sequences that have lower execution cost. Execution cost is
+ * measured by the number of method calls in a sequence and the time it takes to execute.
+ *
+ * <p>Our implementation of Orienteering differs from that described in the GRT paper in that we do
+ * not keep track of the execution time of an input sequence for each new run. Instead, we assume
+ * that an input sequence's execution time is equal to the execution time of its first run. We
+ * believe this assumption is reasonable since a sequence does not take any inputs (it is
+ * self-contained), so its execution time probably does not differ greatly between separate runs.
+ *
+ * <p>The GRT paper also does not describe how to handle input sequences that have an execution time
+ * of zero, such as one that only includes the assignment of a primitive type {@code byte byte0 =
+ * (byte)1;}. We assign these input sequences an execution time of 1 nanosecond to prevent division
+ * by zero when later computing weights.
+ */
+public class OrienteeringSelection extends InputSequenceSelector {
+  /** Map from a sequence to its weight. */
+  private final Map<Sequence, Double> weightMap = new HashMap<>();
+
+  /** Map from a sequence to the number of times it was selected. */
+  private final Map<Sequence, Integer> sequenceSelectionCount = new HashMap<>();
+
+  /**
+   * Cache for the square root of the number of method calls in a sequence of statements. Once
+   * computed for a given sequence, the value is never updated.
+   */
+  private final Map<Sequence, Double> sequenceMethodSizeSqrt = new HashMap<>();
+
+  /**
+   * Map from a sequence to its execution time in nanoseconds. Once computed for a given sequence,
+   * the value is never updated.
+   */
+  private final Map<Sequence, Long> sequenceExecutionTime = new HashMap<>();
+
+  /**
+   * Bias input selection towards lower-cost sequences. We first compute and update the weights of
+   * all the candidates within the candidate list before making our selection.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  @Override
+  public Sequence selectInputSequence(SimpleList<Sequence> candidates) {
+    double totalWeight = computeWeightForCandidates(candidates);
+
+    Sequence selectedSequence = Randomness.randomMemberWeighted(candidates, weightMap, totalWeight);
+    CollectionsPlume.incrementMap(sequenceSelectionCount, selectedSequence);
+
+    // Compute and update the weight of the selected sequence which will be affected by its
+    // increased selection count.
+    double updatedWeight = computeWeightForCandidate(selectedSequence);
+    weightMap.put(selectedSequence, updatedWeight);
+
+    return selectedSequence;
+  }
+
+  /**
+   * Compute the weights of the candidates in the given list that have not been assigned a weight
```

Comment:
How can it happen that some candidate hasn't been assigned a weight?

Reply:
I'm thinking of a situation where Orienteering first chooses from a candidate list of Type A objects. The seed sequences would have no execution times and Orienteering would assign them a weight.

However, later if Orienteering is to choose from a candidate list of Type B objects, there could still be elements in the list (seed sequences) that Orienteering hasn't seen before.